### ISSUE TITLE
Keybindings not working correctly

### ISSUE THREAD

**Issue by FroztOfficial** [qtile/qtile#3856]
### The issue:

Hey, so I just updated to Qtile 0.22.1, but now there is an issue with my config.py.
So I did some debugging an I found out that if there is two keybinding in the keys list that contains a keybinding with "mod", and "mod + mod5", then both the keybindings will be set to the first. Here is an example:
```
mod = "mod4"
altgr = "mod5"

keys = [
  ...
  Key([mod], "p", lazy.spawn(".config/dmenu/dm-utils")),
  Key([mod, altgr], "p", lazy.spawn(".config/dmenu/dm-help")),
]
```
So if this is inside the config.py, then if I do either of the key presses, then it will run this every time:
`lazy.spawn(".config/dmenu/dm-utils")`

### Required:

- [X] I have searched past issues to see if this bug has already been reported.

**Comment by elParaguayo** [comment]
Did that issue not exist in 0.21? If not, we'd need to do a `git bisect` to find the cause of this. Weird!

**Comment by FroztOfficial** [comment]
I just went back to a backup yesterday. And the previous version was "0.21.0", where it worked perfectly.

**Comment by elParaguayo** [comment]
Great. Thanks for confirming.

**Comment by elParaguayo** [comment]
I assume this is on X11, not Wayland. Is that right?

**Comment by FroztOfficial** [comment]
I am using X11.

**Comment by elParaguayo** [comment]
OK - have bisected. This is caused by 3af54868

Pinging @m-col as this was one of his.

**Comment by FroztOfficial** [comment]
Great! I hope it can be fixed soon. I might temporarily use the previous version.

**Comment by m-col** [comment]
That is unfortunate. Please could you post the output of `xev` from pressing the mod4 and mod5 keys (separately)?

**Comment by FroztOfficial** [comment]
Yes, this is the output of the `xev` command, when I press mod4(left windows key), and mod5(altgr)

```
KeyPress event, serial 35, synthetic NO, window 0x2400001,
    root 0x1ee, subw 0x0, time 305163, (34,522), root:(3566,574),
    state 0x0, keycode 133 (keysym 0xffeb, Super_L), same_screen YES,
    XLookupString gives 0 bytes:
    XmbLookupString gives 0 bytes:
    XFilterEvent returns: False

KeyRelease event, serial 35, synthetic NO, window 0x2400001,
    root 0x1ee, subw 0x0, time 305268, (34,522), root:(3566,574),
    state 0x40, keycode 133 (keysym 0xffeb, Super_L), same_screen YES,
    XLookupString gives 0 bytes:
    XFilterEvent returns: False

KeyPress event, serial 35, synthetic NO, window 0x2400001,
    root 0x1ee, subw 0x0, time 305742, (34,522), root:(3566,574),
    state 0x0, keycode 108 (keysym 0xfe03, ISO_Level3_Shift), same_screen YES,
    XKeysymToKeycode returns keycode: 92
    XLookupString gives 0 bytes:
    XmbLookupString gives 0 bytes:
    XFilterEvent returns: False

KeyRelease event, serial 35, synthetic NO, window 0x2400001,
    root 0x1ee, subw 0x0, time 305782, (34,522), root:(3566,574),
    state 0x80, keycode 108 (keysym 0xfe03, ISO_Level3_Shift), same_screen YES,
    XKeysymToKeycode returns keycode: 92
    XLookupString gives 0 bytes:
    XFilterEvent returns: False
```

**Comment by joe-batt** [comment]
I appear to have the same problem, but I am not entirely sure.
I can reproduce this using 0.22.1 (from Frostyx repo) with the default config on Fedora 37 (Python 3.11) on X11. The default bindings bindings for switch to group (M-<number>) and move window to group (M-S-<number>) overlap. Whichever is defined last wins for both bindings.

The weird thing is: I set up a new Fedora 37 VM for some testing so I don't break anything on my machine by accident. I installed qtile the same way and it works perfectly. So far I could not figure out any relevant differences (no differing versions in python packages).

I don't mind rummaging through the code, if you have any ideas on how I could produce more data.

**Comment by joe-batt** [comment]
I can now confirm that my problem is something else. The differentiating is the DisplayManager. On my PC I use SDDM, in the VM I used the default GDM. I can consistently reproduce the overlapping key bindings when using SDDM, with GDM everything is fine. I might open a new issue for this.
